src/Checkin.test.js

```diff
@@ -0,0 +1,738 @@
+
+        fireEvent.click(patronDetailsButton);
+
+        expect(basicProps.mutator.query.update).toHaveBeenCalledWith({
+          _path: `/users/view/${loan.userId}`,
+        });
+      });
+
+      it('should update query pass after clicking on item details button', () => {
+        const itemDetailsButton = screen.getByText(labelIds.itemDetailsButton);
+
+        fireEvent.click(itemDetailsButton);
+
+        expect(basicProps.mutator.query.update).toHaveBeenCalledWith({
+          _path: `/inventory/view/${loan.item.instanceId}/${loan.item.holdingsRecordId}/${loan.item.id}`,
+        });
+      });
+
+      it('should update query pass after clicking on request details button', () => {
+        const requestDetailsButton = screen.getByText(labelIds.requestDetailsButton);
+
+        fireEvent.click(requestDetailsButton);
+
+        expect(basicProps.mutator.query.update).toHaveBeenCalledWith({
+          _path: `/requests/view/${loan.staffSlipContext.request.requestID}`,
+        });
+      });
+
+      it('should trigger get request to receive user information', () => {
+        const newFeeFineButton = screen.getByText(labelIds.newFeeFineButton);
+
+        fireEvent.click(newFeeFineButton);
+
+        expect(basicProps.mutator.users.GET).toHaveBeenCalledWith({
+          params: {
+            query: `id=${loan.userId}`,
+          },
+        });
+      });
+
+      it('should update query pass after clicking on new fee fine button', async () => {
+        const newFeeFineButton = screen.getByText(labelIds.newFeeFineButton);
+
+        fireEvent.click(newFeeFineButton);
+
+        await waitFor(() => {
+          expect(basicProps.mutator.query.update).toHaveBeenCalledWith({
+            _path: `/users/view/${loan.userId}?filters=pg.${groupName}&layer=charge&loan=${loan.id}`,
+          });
+        });
+      });
+
+      it('should show checkin notes after clicking on checkin notes button', () => {
+        const checkinNotesButton = screen.getByText(labelIds.checkinNotesButton);
+
+        fireEvent.click(checkinNotesButton);
+
+        expect(basicProps.showCheckinNotes).toHaveBeenCalledWith(loan);
+      });
+
+      it('should render action menu trigger', () => {
+        const actionMenuTrigger = screen.getByTestId(testIds.actionMenuTrigger);
+
+        expect(actionMenuTrigger).toBeInTheDocument();
+      });
+    });
+
+    describe('When all data for buttons visibility are not set', () => {
+      const loan = {
+        item: {
+          circulationNotes: [],
+          instanceId: 'instanceId',
+          holdingsRecordId: 'holdingsRecordId',
+          id: 'itemId',
+        },
+        id: 'loanId',
+        staffSlipContext: {
+          request: {},
+        },
+      };
+      const props = {
+        ...basicProps,
+        scannedItems: [
+          {
+            item: {},
+          }
+        ]
+      };
+
+      beforeEach(() => {
+        isDCBItem.mockReturnValue(true);
```

Comment:
Although you are mocking the method isDCBItem, and the test are passing successfully, the pre-conditions established are not correct for DCB scenarios..

Just to give you some background on DCB wrt to Checkin app - 
_An item is called as a DCB item when its instanceid and holdingsRecordId are some specific values - refs - https://github.com/folio-org/ui-checkin/blob/c496d3adb3bf0a89d9f584a4b83dc9d9e8ecd1a5/src/consts.js#L66, https://github.com/folio-org/ui-checkin/blob/c496d3adb3bf0a89d9f584a4b83dc9d9e8ecd1a5/src/consts.js#L67, https://github.com/folio-org/ui-checkin/blob/c496d3adb3bf0a89d9f584a4b83dc9d9e8ecd1a5/src/util.js#L174
When it is DCB Item, only  Item Details button is hidden. Refs - https://github.com/folio-org/ui-checkin/pull/608
DCB User-(Also called virtual user)
A user is called as dcb user when his user type is dcb and lastname is DcbSystem
For a DCB user, only New fee/fine button is hidden. Refs - https://github.com/folio-org/ui-checkin/pull/606
The other buttons or menu items in action menu are rendered as is.
Your tests in test suite named - "When all data for buttons visibility are not set" are contradicting and not establishing the correct pre conditions for the tests._

It is probably a good idea to have separate test suites that match the scenario - for instance, have a separate test suite for DCB item and DCB user

Reply:
You left a comment about external isDCBItem method which is placed in another file and it is mocked in my test. I just mocked it and that's it. It does not matter how it is implemented and what DCB item or DCB user means I just need a few different values (true or false) to test all my tests scenarios.
When isDCBItem returns true, item details button should not be visible, when it returns false button should be visible. That's all what I need.
We do not need to rely on inner implementation of isDCBItem and isDcbUser methods inside the test.